### ISSUE TITLE
A few questions on the use

### ISSUE THREAD

**Issue by i-panov** [yii2tech/file-storage#17]
Unfortunately, you do not have a detailed wiki, and on the README not everything is clear.

1. By default, yii2 uses the `yii\web\UploadedFile` class to load files. How do I use it with this library? For example, it has a `saveAs` method. It already implements the storage logic. And of course he does not use this library. Should I not use it at all?

2. There is still a method `yii2tech\filestorage\DownloadAction`, but I did not really understand how to use it... There's not a single example. Only in actions one line. And what's inside? And how do I make a form on the front? How to name the field to work? In general, it would be nice to have another widget for the frontend, as for example in this [library](https://github.com/trntv/yii2-file-kit). In general, for example, there are forms when some more text fields are loaded with the file. So this action does not always work.

3. It is not shown at all whether it is possible to set a key file when connecting to ssh.

**Comment by klimov-paul** [comment]
> 1. By default, yii2 uses the yii\web\UploadedFile class to load files. How do I use it with this library? 

See [yii2tech/ar-file](https://github.com/yii2tech/ar-file) for the example.

> 2. There is still a method yii2tech\filestorage\DownloadAction, but I did not really understand how to use it...

It is a substitution for the native bucket file download mechanism
See [Bucket::getFileUrl() docs](https://github.com/yii2tech/file-storage#accessing-files-by-url-) - I can not explain more detailed.

> 3. It is not shown at all whether it is possible to set a key file when connecting to ssh.

See https://github.com/yii2tech/file-storage/blob/master/src/sftp/Connection.php#L56-L80

**Comment by i-panov** [comment]
@klimov-paul Thanks for the answer. What about the limit on the number of files in the folder? I have a task to make for example that no more than 1000 files are saved in one folder. That is, there are subfolders with random names (uniqid). When there are more than 1000 files in the folder, a new folder with a random name is created and the files are saved already in it. It is considered that when there are too many files in one folder, it can brake... This is not my idea, but I need to do this. I did not find a way to view the list of files on a remote server using this library.

For example, in [this](https://github.com/trntv/yii2-file-kit/blob/master/src/Storage.php) the library does this almost as I described, but instead of the unique identifiers, the sequence numbers of the folders.
